### PR TITLE
Added Sum Of Divisors in CPP

### LINKED ISSUE #4730: Sum of Divisors of a Number
## 💥 Proposal

(A clear and concise description of what the proposal is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

YES.

### PR DESCRIPTION
**Pull request by RaiyanMahin**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

Given a Number. Find it's Number of Divisors.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #4730

### CODE REVIEW

File: C-Plus-Plus/math/sum_of_divisors.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    for(int i = 1; i < limit; i++)
+    {
+        if(Number % i == 0)
+        {
+            /* Here i is a divisor of that number
+            ( Number / i ) is also a divisor
+            Ex : 16
+            16 / 2 == 8
+            8 , 2 both are divisor of 16
+             */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/sum_of_divisors.cpp]: Remove the redundant comments

File: C-Plus-Plus/math/sum_of_divisors.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            sum_of_divisors += i;
+            sum_of_divisors += (Number / i);
+        }
+    }
+    if(Number % limit == 0)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/sum_of_divisors.cpp:32]: Add the new line above the if condition

File: C-Plus-Plus/math/sum_of_divisors.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        if(limit * limit == Number)
+        {
+            /* means perfect square number
+            Ex : 16
+            4 * 4 == 16
+            */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/sum_of_divisors.cpp]: Remove the redundant comments

File: C-Plus-Plus/math/sum_of_divisors.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            sum_of_divisors += limit;
+        }
+        else
+        {
+            /* Here i is a divisor of that number
+            ( Number / i ) is also a divisor
+            */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/sum_of_divisors.cpp]: Remove the redundant comments

File: C-Plus-Plus/math/sum_of_divisors.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+int main()
+{
+        cout << "Enter the number : \n";
+	int Number;
+	cin >> Number;
+	long long int sum_of_divisors_of_the_number = get_sum_of_divisors_of_the_number(Number);
+	cout << "Sum of Divisors of this Number is : ";
+	cout << sum_of_divisors_of_the_number << endl;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/sum_of_divisors.cpp]: Align your code properly also `sum_of_divisors_of_the_number` rename this to `sum_of_divisors`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by RaiyanMahin** [comment]
all done!

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**